File: src/song/router.ts
import { Router } from 'express';

import {
  handleGetSongs,
  handleGetSong,
  handleCreateSong,
  handleUpdateSong,
  handleDeleteSong,
} from './controller';
import authMiddleware from '../auth/middleware';

const songRouter = Router();

songRouter.get('/', handleGetSongs);
songRouter.get('/:id', handleGetSong);
songRouter.post('/', authMiddleware, handleCreateSong);
songRouter.put('/:id', authMiddleware, handleUpdateSong);
songRouter.delete('/:id', authMiddleware, handleDeleteSong);

export default songRouter;

File: songs/router.go
package songs

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitSongsRouter(r *mux.Router) {
	r.HandleFunc("/songs", handleGetSongs).Methods("GET")

	r.HandleFunc("/songs/{id}", auth.VerifyAuthorization(handleGetSongByID)).Methods("GET")
	r.HandleFunc("/songs", auth.VerifyAuthorization(handlePostSong)).Methods("POST")
	r.HandleFunc("/songs/{id}", auth.VerifyAuthorization(handlePatchSong)).Methods("PATCH")
	r.HandleFunc("/songs/{id}", auth.VerifyAuthorization(handleDeleteSong)).Methods("DELETE")
}
